### PR TITLE
PBENCH-697 Delete stockpile submodule

### PR DESCRIPTION
**Pull request by ndokos**
PBENCH-697

There are currently two commits: the first one gets rid of stockpile uses and references everywhere except the test output; the second fixes the tests.

Contents of the first commit:

- Delete the git submodule.

- Delete the stockpile patch file and clean up the agent spec file.

- Delete stockpile collection function and reference in
  pbench-syssinfo-dump.

- Delete references in pbench-run-benchmark.pl and
  lib/PbenchAnsbile.pm and delete comment in Systat.pm.

- Delete references in docs/CONFIG_DATA.md.

- Delete reference in lib/pbench/agent/constants.py.

- Delete references in build.sh and pyproject.toml.

- Delete references in agent/Makefile.

- Clean up agent/base.

- Clean up .flake8 and .pre-commit-config.yaml.

Diffstat of the second commit:

 - agent/bench-scripts/tests/pbench-fio/test-14.txt            |  2 +-
 - agent/bench-scripts/tests/pbench-fio/test-17.txt            |  2 +-
 - agent/bench-scripts/tests/pbench-fio/test-18.txt            |  2 +-
 - agent/bench-scripts/tests/pbench-fio/test-21.txt            |  2 +-
 - agent/bench-scripts/tests/pbench-fio/test-26.txt            |  2 +-
 - agent/bench-scripts/tests/pbench-fio/test-27.txt            |  2 +-
 - agent/bench-scripts/tests/pbench-fio/test-29.txt            |  2 +-
 - agent/bench-scripts/tests/pbench-uperf/test-59.txt          |  2 +-
 - agent/bench-scripts/tests/pbench-uperf/test-60.txt          |  2 +-
 - agent/bench-scripts/tests/pbench-user-benchmark/test-33.txt |  2 +-
 - agent/bench-scripts/tests/pbench-user-benchmark/test-34.txt |  2 +-
 - agent/bench-scripts/tests/pbench-user-benchmark/test-35.txt |  2 +-
 - agent/bench-scripts/tests/pbench-user-benchmark/test-36.txt |  2 +-
 - agent/bench-scripts/tests/test-benchmark-clis/test-CL.txt   | 30 +++++++++++++++---------------

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 # black complains loudly that fio-histo-log-pctile.py is not a python file
```
Comment (dbutenhof) [inline .pre-commit-config.yaml:2]: Funny that the comment justifies one of three entries here, but then it used to be one of four. 😁

**Review by dbutenhof — APPROVED** [review]
I'm disabling legacy tests, but that PR still has a ways to go. I guess I should "fix" (snip, snip) the test script in a separate PR to get that through quicker. 😆

File: .pre-commit-config.yaml (hunk 1 above)
Comment (ndokos) [reply .pre-commit-config.yaml:2]: As @webbnh says: "Every day, in every way, I'm getting better and better...." [1]  - eventually there will only be one entry and the comment will be correct :-)

[1] Of course, he was quoting Inspector Clouseau, so maybe... Never mind!

Comment (dbutenhof) [reply .pre-commit-config.yaml:2]: Ha. Meaning, we'll never be done. Job security.

Comment (webbnh) [reply .pre-commit-config.yaml:2]: > eventually there will only be one entry and the comment will be correct :-)

I just put up PR #2972, which, once the conflict with this PR is resolved, will have that very effect...

> he was quoting Inspector Clouseau

Actually, that was Commissioner Dreyfus, Clouseau's boss....

**Review by webbnh — APPROVED** [review]
👍

**Pull request merged**